[InstagramBridge] Documentation for configuring this bridge (#2437)
RSS-Bridge/rss-bridge: doc/bridges/InstagramBridge.rst

=================
 InstagramBridge
=================

> To somehow bypass `rate limiting issue <https://github.com/RSS-Bridge/rss-bridge/issues/1891>`_,

Reviewer: ```suggestion
To somehow bypass the `rate limiting issue <https://github.com/RSS-Bridge/rss-bridge/issues/1891>`_,
```

> it is suggested to deploy private RSS-Bridge that uses working Instagram account.

Reviewer: ```suggestion
it is suggested to deploy a private RSS-Bridge instance that uses a working Instagram account.
```

Configuration
-------------

1. Retreiving session id.

> Following steps describe how to get it using Chromium based browser.

Reviewer: ```suggestion
The following steps describe how to get the session id using a Chromium-based browser.
```

> - Create Instagram account, that you will use for your RSS-Bridge instance.

Reviewer: ```suggestion
- Create an Instagram account, that you will use for your RSS-Bridge instance.
```

> It is NOT recommended to use existing account, that is used for common interaction with Instagram services.

Reviewer: ```suggestion
It is NOT recommended to use your existing account that is used for common interaction with Instagram services.
```

- Login to Instagram

- Open DevTools by pressing F12

- Open "Networks tab"

> - In "Filter" field input "i.instagram.com"

Reviewer: ```suggestion
- In the "Filter" field input "i.instagram.com"
```

- Click on "Fetch/XHR"

- Refresh web page

> - Click on any item from table with http requests

Reviewer: ```suggestion
- Click on any item from the table of http requests
```

> - In new frame open "Headers" tab and scroll to "Request headers"

Reviewer: ```suggestion
- In the new frame open the "Headers" tab and scroll to "Request Headers"
```

> - There will be cookie param will lots of "key=value" text. You need the value of "sessionid" key. Copy it

Reviewer: ```suggestion
- There will be a cookie param will lots of ``<key>=<value>;`` text. You need the value of the "sessionid" key. Copy it.
```

> 2. Configurating RSS-Bridge

Reviewer: ```suggestion
2. Configuring RSS-Bridge
```

- In config.ini.php add following configuration:

.. code-block::

   [InstagramBridge]
   session_id = %sessionid from step 1%
   cache_timeout = %cache timeout in seconds%

> The more cache_timeout value, less chances for RSS-Bridge to throw 429 errors.

Reviewer: ```suggestion
The bigger the cache_timeout value, the smaller the chance for RSS-Bridge to throw 429 errors.
```
